### PR TITLE
Remove enzyme

### PR DESCRIPTION
**Pull request by juanpicado**
**Type:** refactor

**Description:**

We did move to testing library but still using enzyme, this PR remove enzyme completely. 

* Remove enzyme (and related tools, extensions etc etc)
* Add new tools for testing library (dom)
* Update Jest

⚠️ None component has changed, only test, snapshots and expectations.

### CODE REVIEW

File: i18n/config.ts [hunk 1]
```diff
@@ -64,20 +63,7 @@
...
-      'de-DE',
-      'fr-FR',
-      'zh-CN',
-      'ja-JP',
-      'ru-RU',
-      'uk-UA',
-      'km-KH',
-      'zh-TW',
-    ],
+    whitelist: ['en-US', 'cs-CZ', 'pt-BR', 'es-ES', 'de-DE', 'fr-FR', 'zh-CN', 'ja-JP', 'ru-RU', 'uk-UA', 'km-KH', 'zh-TW'],
```
Comment (sergiohgz) [inline i18n/config.ts]: This seems a Prettier problem, is working properly?

File: src/components/CopyToClipBoard/CopyToClipBoard.test.tsx [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
+import { render, cleanup, fireEvent } from 'verdaccio-ui/utils/test-react-testing-library';
 
 import CopyToClipBoard from './CopyToClipBoard';
-import { CopyIcon } from './styles';
 
 jest.mock('verdaccio-ui/utils/cli-utils');
 
 describe('<CopyToClipBoard /> component', () => {
-  let wrapper: ReactWrapper;
+  let wrapper: any;
```
Comment (sergiohgz) [inline src/components/CopyToClipBoard/CopyToClipBoard.test.tsx:11]: Why this type is any now?

File: src/components/CopyToClipBoard/CopyToClipBoard.test.tsx [hunk 3]
```diff
@@ -1,28 +1,27 @@
...
   const copyText = 'copy text';
 
   beforeEach(() => {
-    wrapper = mount(<CopyToClipBoard text={copyText} />);
+    cleanup();
```
Comment (sergiohgz) [inline src/components/CopyToClipBoard/CopyToClipBoard.test.tsx:15]: Testing Library says that we should not perform manual cleanup because of its own internal mechanism, except some cases. Is this one of the corner cases??

File: src/components/CopyToClipBoard/__snapshots__/CopyToClipBoard.test.tsx.snap [hunk 4]
```diff
@@ -1,3 +1,182 @@
...
+            />
+          </svg>
+        </span>
+        <span
+          class="MuiTouchRipple-root"
+        />
+      </button>
+    </div>
+  </div>,
+  "debug": [Function],
```
Comment (sergiohgz) [inline src/components/CopyToClipBoard/__snapshots__/CopyToClipBoard.test.tsx.snap:130]: This snapshot seems not be correct, it's writing everything method returned by render function. Should we write container only?

**Review by sergiohgz — COMMENTED** [review]
Some important comments to take care of them

File: i18n/config.ts (hunk 1 above)
Comment (juanpicado) [reply i18n/config.ts]: my bad :) don't use `--no-verify` 😋

File: src/components/CopyToClipBoard/CopyToClipBoard.test.tsx (hunk 2 above)
Comment (juanpicado) [reply src/components/CopyToClipBoard/CopyToClipBoard.test.tsx:11]: Not sure yet, I just wanna remove enzyme. We will figure out.

File: src/components/CopyToClipBoard/CopyToClipBoard.test.tsx (hunk 3 above)
Comment (juanpicado) [reply src/components/CopyToClipBoard/CopyToClipBoard.test.tsx:15]: Only if you enable in the setup, not yet enabled, we need to review all test first.

File: src/components/CopyToClipBoard/__snapshots__/CopyToClipBoard.test.tsx.snap (hunk 4 above)
Comment (juanpicado) [reply src/components/CopyToClipBoard/__snapshots__/CopyToClipBoard.test.tsx.snap:130]: Maybe, I just would like remove all snapshots, as you say, are not helping. For now I just migrated to testing library. Follow ups I'd like review all tests, too many are using snapshots.

**Comment by DanielRuf** [comment]
Nothing to add to the comments which were already given. I guess this is still WIp due to the breaking builds?

**Comment by juanpicado** [comment]
> Nothing to add to the comments which were already given. I guess this is still WIP due to the breaking builds?

I forgot push. All good now.  The code analysis reflect duplicate code but does not make much sense on test files, I already changed it and make it works. I also excluded `*.test.*` from duplication but still insist. I'll check that in the future.

**Comment by sergiohgz** [comment]
> > Nothing to add to the comments which were already given. I guess this is still WIP due to the breaking builds?
> 
> I forgot push. All good now.  The code analysis reflect duplicate code but does not make much sense on test files, I already changed it and still complains, is ridiculous strict. I also excluded `*.test.*` from duplication but still insist. I'll check that in the future. 

This settings are done in SonarCloud or directly in a properties file? I'll check it

**Comment by juanpicado** [comment]
> > > Nothing to add to the comments which were already given. I guess this is still WIP due to the breaking builds?
> > 
> > 
> > I forgot push. All good now.  The code analysis reflect duplicate code but does not make much sense on test files, I already changed it and still complains, is ridiculous strict. I also excluded `*.test.*` from duplication but still insist. I'll check that in the future.
> 
> This settings are done in SonarCloud or directly in a properties file? I'll check it

I tried to update settings at SonarCloud but I don't see the changes reflected. Perhaps you can check.

**Comment by DanielRuf** [comment]
> > > Nothing to add to the comments which were already given. I guess this is still WIP due to the breaking builds?
> > 
> > 
> > I forgot push. All good now.  The code analysis reflect duplicate code but does not make much sense on test files, I already changed it and still complains, is ridiculous strict. I also excluded `*.test.*` from duplication but still insist. I'll check that in the future.
> 
> This settings are done in SonarCloud or directly in a properties file? I'll check it

Basically yes. I didn't touch a sonarcloud / sonarscanner config in a while.

**Review by frimuchkov — APPROVED** [review]

**Comment by juanpicado** [comment]
Ok, we are good to go then. :)

**Comment by juanpicado** [comment]
Thanks @frimuchkov 💯 for the review !
